syntax = "proto3";

service Logs {
    rpc FetchLogs (LogsRequest) returns (LogsReply) {}
    rpc SearchLogs (LogsRequest) returns (LogsReply) {}
}

message LogsRequest {
    optional int32 from_log = 1;
    optional int32 to_log = 2;
    optional string search_text = 3;
}

message LogsReply {
    message Log {
        string context = 1;
        string created = 2;
        int32 id = 3;
        string level = 4;
        string text = 5;
    }
    repeated Log logs = 1;
}